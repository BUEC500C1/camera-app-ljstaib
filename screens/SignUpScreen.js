//Luke Staib 2021 ljstaib@bu.edu
//Sign Up Screen

import React, { useState } from 'react';
import { Keyboard, Image, Text, TextInput, TouchableOpacity, View } from 'react-native';
import { Styles } from '../Styles';

import * as firebase from 'firebase';

function SignUpScreen({ navigation }) {
  const [email, setEmail] = useState(null);
  const [password, setPassword] = useState(null);
  
  const resetVars = () => { //Reset fields upon leaving screen
    setEmail(null);
    setPassword(null);
  }
  
  return (
    <View style={Styles.container}>
      <Text style={Styles.app_title_marginbottom}>Sign Up</Text>
      <TextInput
          autoCapitalize = "none"
          autoCompleteType = "off"
          autoCorrect = {false}
          maxLength = {32}
          onChangeText = {email => setEmail(email)}
          onSubmitEditing = {()=> Keyboard.dismiss()}
          placeholder = "Email"
          style = {Styles.auth_input_text}
          value = {email}
      />
      <Text style={Styles.small_text}>Passwords must be at least 8 characters long.</Text>
      <TextInput
          autoCapitalize = "none"
          autoCompleteType = "off"
          autoCorrect = {false}
          maxLength = {32}
          onChangeText = {password => setPassword(password)}
          onSubmitEditing = {()=> Keyboard.dismiss()}
          placeholder = "Password"
          secureTextEntry
          style = {Styles.auth_input_text}
          value = {password}
      />
      <TouchableOpacity onPress = {() => {
          if (signUpEmailPass(navigation, email, password) === true) {
            resetVars();
          }
        }}
        style={Styles.auth_button}>
        <Text style={Styles.white_text}>Sign Up</Text>
      </TouchableOpacity>
      <TouchableOpacity onPress = {() => {resetVars(); navigation.navigate('LoginScreen');}} style={Styles.auth_button}>
        <Text style={Styles.white_text}>Back</Text>
      </TouchableOpacity>
    </View>
  );
}

async function signUpEmailPass(navigation, email, password) {
  console.log("INFO: Sign up button pressed")
  if (email !== null && password.length >= 8) {
    firebase.auth().createUserWithEmailAndPassword(email, password)
    .then(() => {
      firebase.auth().currentUser.sendEmailVerification()
      .then(() => {
        alert("User created successfully. Please verify your email before logging in.")
        navigation.navigate('LoginScreen')
        return true
      })
      .catch((e) => {
        console.log("ERROR: Code " + String(e.code) + " with message " + String(e.message))
        alert("Could not send email verification.")
        return false
      })
    })
    .catch((e) => {
      console.log("ERROR: Code " + String(e.code) + " with message " + String(e.message))
      if (e.code == "auth/email-already-in-use") {
        alert("This email already has an account assigned to it.")
      }
      else if (e.code == "auth/invalid-email") {
        alert("Please enter a valid email address.")
      }
      else {
        alert("Could not sign up due to an unknown error. Please try again later.")
      }
      return false
    });
  }
  else {
    alert("Please enter an email and password. Passwords must be at least 8 characters long.")
    return false
  }
}

export default SignUpScreen;
