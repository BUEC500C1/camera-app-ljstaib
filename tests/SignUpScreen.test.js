//Luke Staib 2021 ljstaib@bu.edu
//Sign Up Testing

import React from 'react';
import renderer from 'react-test-renderer';

import SignUpScreen from '../screens/SignUpScreen';

describe('<SignUpScreen />', () => {
  //Test if rendering is correct
  it('renders correctly', () => {
    const tree = renderer.create(<SignUpScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
