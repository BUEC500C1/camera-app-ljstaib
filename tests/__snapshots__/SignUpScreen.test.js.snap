// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffed",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontFamily": "Helvetica-Bold",
        "fontSize": 48,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Sign Up
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCompleteType="off"
    autoCorrect={false}
    maxLength={32}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 2,
        "color": "black",
        "fontFamily": "Helvetica",
        "fontSize": 24,
        "marginBottom": 10,
        "textAlign": "center",
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value={null}
  />
  <Text
    style={
      Object {
        "color": "black",
        "fontFamily": "Helvetica",
        "fontSize": 16,
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Passwords must be at least 8 characters long.
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCompleteType="off"
    autoCorrect={false}
    maxLength={32}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 2,
        "color": "black",
        "fontFamily": "Helvetica",
        "fontSize": 24,
        "marginBottom": 10,
        "textAlign": "center",
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value={null}
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#ffe29c",
        "borderRadius": 30,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "padding": 1,
        "width": "60%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 32,
          "textAlign": "center",
        }
      }
    >
      Sign Up
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#ffe29c",
        "borderRadius": 30,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "padding": 1,
        "width": "60%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 32,
          "textAlign": "center",
        }
      }
    >
      Back
    </Text>
  </View>
</View>
`;
